### PR TITLE
feat(plugin): bundle analyze supports for rspack's banner plugin

### PR DESCRIPTION
**Pull request by easy1090**
## Summary
When BannerPlugin is used in the rspack project to add templates code to the assets, it will cause Rsdoctor's Bundle parsing to fail. In summary, Rsdoctor bundle analysis is required to support Banner Plugin.

```js
// rspack.config.js

plugins: [
    new rspack.BannerPlugin({
      test: /\.js/,
      banner,
      raw: true,
    }),

]

```

## Related Links
 closes: #345

### CODE REVIEW

File: e2e/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "name": "@rsdoctor/e2e",
   "version": "0.0.1",
   "scripts": {
-    "test": "playwright test"
+    "test": "playwright test --update-snapshots"
```
Comment (chenjiahan) [inline e2e/package.json:6]: should we add a `test:update` script for this?

I'm not sure if this change will make the E2E CI to update snapshot too.

File: examples/rspack-banner-minimal/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# cra-app
```
Comment (chenjiahan) [inline examples/rspack-banner-minimal/CHANGELOG.md:1]: unused changelog?

File: examples/rspack-banner-minimal/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import '@arco-design/web-react/dist/css/arco.css';
+import styles from './index.module.less';
+
+export function App({ name }: any) {
+  return (
+    <>
+      Hello <i>{name}</i>. Welcome!
+      <>
+        <h1 className={styles.header}>标题{name}</h1>
+        <p className="worker">内容</p>
```
Comment (chenjiahan) [inline examples/rspack-banner-minimal/src/App.tsx:12]: English by default

File: packages/core/src/build-utils/build/utils/parseBundle.ts [hunk 4]
```diff
@@ -31,7 +32,27 @@ export const parseBundle: ParseBundle = (
     return {};
   }
 
-  const content = fs.readFileSync(bundlePath, 'utf8');
+  let content = fs.readFileSync(bundlePath, 'utf8');
+  const tagCache = new Map();
+
+  let _hasBannerPlugin = content.indexOf('RSDOCTOR_START::') > 0;
```
Comment (chenjiahan) [inline packages/core/src/build-utils/build/utils/parseBundle.ts:38]: Can we just rename `_hasBannerPlugin` to `hasBannerPlugin`?

File: packages/core/src/build-utils/build/utils/parseBundle.ts [hunk 5]
```diff
@@ -31,7 +32,27 @@ export const parseBundle: ParseBundle = (
...
+
+  if (_hasBannerPlugin && !tagCache.get(bundlePath)) {
+    const tagMatchResult = getStringBetween(
+      content,
+      0,
+      /([a-z|A-Z]+\.[a-z]+)\(\SRSDOCTOR_START::(.*?);/,
```
Comment (chenjiahan) [inline packages/core/src/build-utils/build/utils/parseBundle.ts:44]: Is `SRSDOCTOR` a typo?

File: examples/rspack-banner-minimal/package.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    "classnames": "^2.5.1",
+    "css-loader": "^6.9.0",
+    "less-loader": "^11.1.4",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
+  },
+  "devDependencies": {
+    "tsm": "2.3.0",
+    "@rsdoctor/rspack-plugin": "workspace:*",
+    "@rspack/cli": "0.5.1",
```
Comment (chenjiahan) [inline examples/rspack-banner-minimal/package.json:23]: We can upgrade Rspack to the latest version (in another PR)

File: examples/rspack-banner-minimal/public/robots.txt
Comment (chenjiahan) [inline examples/rspack-banner-minimal/public/robots.txt:1]: Unused file

File: examples/rspack-banner-minimal/package.json (hunk 6 above)
Comment (easy1090) [reply examples/rspack-banner-minimal/package.json]: Updated rsbuild and rspack.

File: e2e/package.json (hunk 1 above)
Comment (easy1090) [reply e2e/package.json]: This is my mistake, i forgot to delete the test code.

File: packages/core/src/build-utils/build/utils/parseBundle.ts (hunk 5 above)
Comment (easy1090) [reply packages/core/src/build-utils/build/utils/parseBundle.ts:44]: This \S is a Regexr.

Comment (chenjiahan) [reply packages/core/src/build-utils/build/utils/parseBundle.ts:44]: Oh, get it..

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**